tests/qxmpptrustmessages/tst_qxmpptrustmessages.cpp

```diff
@@ -0,0 +1,214 @@
+/*
+ * Copyright (C) 2008-2021 The QXmpp developers
+ *
+ * Authors:
+ *  Melvin Keskin
+ *
+ * Source:
+ *  https://github.com/qxmpp-project/qxmpp
+ *
+ * This file is a part of QXmpp library.
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ */
+
+#include <QObject>
+
+#include "QXmppTrustMessageElement.h"
+#include "QXmppTrustMessageKeyOwner.h"
+
+#include "util.h"
+
+class tst_QXmppTrustMessages : public QObject
+{
+    Q_OBJECT
+
+private slots:
+    void testIsTrustMessageKeyOwner_data();
+    void testIsTrustMessageKeyOwner();
+    void testTrustMessageKeyOwner_data();
+    void testTrustMessageKeyOwner();
+    void testIsTrustMessageElement_data();
+    void testIsTrustMessageElement();
+    void testTrustMessageElement();
+};
+
+void tst_QXmppTrustMessages::testIsTrustMessageKeyOwner_data()
+{
+    QTest::addColumn<QByteArray>("xml");
+    QTest::addColumn<bool>("isValid");
+
+    QTest::newRow("valid")
+        << QByteArrayLiteral("<key-owner xmlns=\"urn:xmpp:tm:0\"/>")
+        << true;
+    QTest::newRow("invalidTag")
+        << QByteArrayLiteral("<invalid xmlns=\"urn:xmpp:tm:0\"/>")
+        << false;
+    QTest::newRow("invalidNamespace")
+        << QByteArrayLiteral("<key-owner xmlns=\"invalid\"/>")
+        << false;
+}
+
+void tst_QXmppTrustMessages::testIsTrustMessageKeyOwner()
+{
+    QFETCH(QByteArray, xml);
+    QFETCH(bool, isValid);
+
+    QDomDocument doc;
+    QCOMPARE(doc.setContent(xml, true), true);
+    const QDomElement element = doc.documentElement();
+    QCOMPARE(QXmppTrustMessageKeyOwner::isTrustMessageKeyOwner(element), isValid);
+}
+
+void tst_QXmppTrustMessages::testTrustMessageKeyOwner_data() {
```

Comment:
please pay attention to the coding style
```suggestion
void tst_QXmppTrustMessages::testTrustMessageKeyOwner_data()
{
```